packages/Interface/src/DTK_C_API.cpp

```diff
@@ -338,108 +358,151 @@ void DTK_destroy( DTK_UserApplicationHandle handle )
+                PolyhedronListSizeFunctionWrapper, data );
+            break;
+        case DTK_POLYHEDRON_LIST_DATA_FUNCTION:
+            dtk->_registry->setPolyhedronListDataFunction(
+                PolyhedronListDataFunctionWrapper, data );
+            break;
+        case DTK_CELL_LIST_SIZE_FUNCTION:
+            dtk->_registry->setCellListSizeFunction( CellListSizeFunctioWrapper,
+                                                     data );
+            break;
+        case DTK_CELL_LIST_DATA_FUNCTION:
+            dtk->_registry->setCellListDataFunction(
+                CellListDataFunctionWrapper, data );
+            break;
+        case DTK_MIXED_TOPOLOGY_CELL_LIST_SIZE_FUNCTION:
+            dtk->_registry->setMixedTopologyCellListSizeFunction(
+                MixedTopologyCellListSizeFunctionWrapper, data );
+            break;
+        case DTK_MIXED_TOPOLOGY_CELL_LIST_DATA_FUNCTION:
+            dtk->_registry->setMixedTopologyCellListDataFunction(
+                MixedTopologyCellListDataFunctionWrapper, data );
+            break;
+        case DTK_BOUNDARY_SIZE_FUNCTION:
+            dtk->_registry->setBoundarySizeFunction(
+                BoundarySizeFunctionWrapper, data );
+            break;
+        case DTK_BOUNDARY_DATA_FUNCTION:
+            dtk->_registry->setBoundaryDataFunction(
+                BoundaryDataFunctionWrapper, data );
+            break;
+        case DTK_DOF_MAP_SIZE_FUNCTION:
+            dtk->_registry->setDOFMapSizeFunction( DOFMapSizeFunctionWrapper,
+                                                   data );
+            break;
+        case DTK_DOF_MAP_DATA_FUNCTION:
+            dtk->_registry->setDOFMapDataFunction( DOFMapDataFunctionWrapper,
+                                                   data );
+            break;
+        case DTK_MIXED_TOPOLOGY_DOF_MAP_SIZE_FUNCTION:
+            dtk->_registry->setMixedTopologyDOFMapSizeFunction(
+                MixedTopologyDOFMapSizeFunctionWrapper, data );
+            break;
+        case DTK_MIXED_TOPOLOGY_DOF_MAP_DATA_FUNCTION:
+            dtk->_registry->setMixedTopologyDOFMapDataFunction(
+                MixedTopologyDOFMapDataFunctionWrapper, data );
+            break;
+        case DTK_FIELD_SIZE_FUNCTION:
+            dtk->_registry->setFieldSizeFunction( FieldSizeFunctionWrapper,
+                                                  data );
+            break;
+        case DTK_PULL_FIELD_DATA_FUNCTION:
+            dtk->_registry->setPullFieldDataFunction(
+                PullFieldDataFunctionWrapper<double>, data );
+            break;
+        case DTK_PUSH_FIELD_DATA_FUNCTION:
+            dtk->_registry->setPushFieldDataFunction(
+                PushFieldDataFunctionWrapper<double>, data );
+            break;
+        case DTK_EVALUATE_FIELD_FUNCTION:
+            dtk->_registry->setEvaluateFieldFunction(
+                EvaluateFieldFunctionWrapper<double>, data );
+        }
```

Comment:
It's arguable if we need it. I'm not sure if it's possible to create an integer and then pass it to a function taking an enum without any issues. I think in C++ you would certainly need a `static_cast`, but not sure about C.